modules/core/src/main/java/org/apache/ignite/internal/processors/cache/checker/processor/AbstractPipelineProcessor.java

```diff
@@ -317,10 +331,16 @@ protected void printStatistics() {
      * Returns a cluster group represents a set od nodes that own the given partition.
      * Returned group can be {@code null} in case of there are no owners for the given partition.
      *
+     * @throws IgniteException If failed to get partition owners.
      * @return Cluster group of owners.
      */
-    private ClusterGroup partOwners(String cacheName, int partId) {
-        Collection<ClusterNode> nodes = ignite.cachex(cacheName).context().topology().owners(partId, startTopVer);
+    private ClusterGroup partOwners(String cacheName, int partId) throws IgniteException {
+        IgniteInternalCache<?, ?> cachex = ignite.cachex(cacheName);
+
+        if (cachex == null)
+            throw new IgniteException("Cache not found (was stopped) [name=" + cacheName + ']');
+
+        Collection<ClusterNode> nodes = cachex.context().topology().owners(partId, startTopVer);
```

Comment:
Do we handle cache owners changes during further task processing?

Reply:
We calculate owners for the given topology. Changing the owners means changing the cluster topology. Therefore, this partition reconciliation session will be canceled.